src/vivarium_public_health/metrics/stratification.py

```diff
@@ -26,50 +26,94 @@ class SourceType(Enum):
 
 @dataclass
 class Source:
-    # todo add docstrings
+    """
+    A source of information about simulants used to determine which category
+    they belong to for a given stratification level. The source name should be
+    the name of the column or pipeline being used as the source. If the source
+    is of type clock, the name should be something descriptive and unique.
+    """
     name: str
     type: SourceType
 
 
 @dataclass
 class StratificationLevel:
-    # todo add docstrings
+    """
+    A level of stratification. Each StratificationLevel represents a set of
+    mutually exclusive and collectively exhaustive categories into which
+    simulants can be assigned.
+
+    The mapper is a function which is applied to the list of sources and
+    assigns each simulant to specific category. It will throw an error if its
+    output is not one of the stratification's categories. By default, the
+    mapper assumes the StratificationLevel has a single source and returns its
+    value as the category.
+
+    A current category getter can be defined if it is known that certain
+    categories are not possible during specific times. Removing these
+    categories from iteration can provide a performance improvement in these
+    cases. The primary use case for the current category getter is for
+    stratification by time, and in particular stratification by year. By
+    default, this will return all categories.
+    """
     name: str
     sources: List[Source]
     categories: Set[str]
     mapper: Callable[[pd.Series], str] = None
     current_categories_getter: Callable[[], Set[str]] = None
 
     def __post_init__(self):
-        self.mapper = self.mapper if self.mapper else self._default_mapper
+        self._set_mapper()
+        self._set_current_categories_getter()
```

Comment:
Do we use self._set_current_categories_getter() instead of using what's in the body of the function to define self.current_categories_getter for consistency between these components? I was thinking we were setting the mapper so that we could wrap it around the category validator but I don't see an equivalent reason to do the same for the current categories getter.

Reply:
There isn't a fundamental reason for why I changed how I'm setting the current categories getter. I just changed it like this to be consistent with the `self._set_mapper()` function